src/otx/algo/diffusion/clip_tokenizer.py

```diff
@@ -0,0 +1,143 @@
+"""This module provides the ClipTokenizer class for tokenizing text using the ClipTokenizer algorithm."""
+
+from __future__ import annotations
+
+import gzip
+import re
+from functools import lru_cache
+
+from .utils.download import download
+
+
+@lru_cache
```

Comment:
Should we have `lru_cache` for this? This looks like called once when creating Tokenizer class.